SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimAnalyzer.cc

```diff
@@ -261,7 +460,522 @@ EcalEBTrigPrimAnalyzer::analyze(const edm::Event& iEvent, const  edm::EventSetup
+      if ( isGenParticleValid_ )  {
+	BaseParticlePropagator prop;
+
+	for ( gpIter=gp->begin() ; gpIter!=gp->end() ; gpIter++ ) {
+	  if ( gpIter->pdgId() == 22 ) {
+	    if (abs(gpIter->eta()) < 1.5 && gpIter->pt()>10 ) {
+
+	      
+	      const reco::GenParticle* myGp=&(*gpIter);
+              RawParticle trueEle(myGp->p4());
+	      trueEle.setVertex(myGp->vertex().x(), myGp->vertex().y(), myGp->vertex().z(), 0.);
+	      if (debug_) std::cout << " Before propagation gen ID " << gpIter->pdgId() << " p " << myGp->p() << " vertex " << myGp->vertex().x() << " " << myGp->vertex().y() << " " << myGp->vertex().z() << std::endl;
+	
+              float ch=gpIter->charge();
+	      if (debug_) std::cout << " my charge " << ch << std::endl;
+	      trueEle.setCharge(ch);
+	      prop = BaseParticlePropagator(trueEle,0.,0.,4.);
+	      BaseParticlePropagator start(prop);
+	      prop.propagateToEcalEntrance();
+              if ( prop.getSuccess() !=0) {
+                trueEle_p4 = new reco::Candidate::PolarLorentzVector(prop.E()*sin(prop.vertex().theta()),  prop.vertex().eta(), prop.vertex().phi(), 0.);
+                float pstart = sqrt ( start.momentum().x()*start.momentum().x() + start.momentum().y()*start.momentum().y() + start.momentum().z()*start.momentum().z()  );
+		if (debug_) std::cout << " starting state   "  << " vertex " << start.vertex().x() << " " << start.vertex().y() << " " << start.vertex().z() << " momentum " << pstart << std::endl;
+                float pprop = sqrt ( prop.momentum().x()*prop.momentum().x() + prop.momentum().y()*prop.momentum().y() + prop.momentum().z()*prop.momentum().z()  );
+		 if (debug_)std::cout << " After propagation "  << " vertex " << prop.vertex().x() << " " << prop.vertex().y() << " " << prop.vertex().z() << " momentum " << pprop << std::endl;
+		if (debug_) std::cout << " Momentum difference " << myGp->p()-pprop << std::endl;
+	      } else {
+		if (debug_) std::cout << " Prop fails " << std::endl;
+                continue;
+	      }
+
+
+	      if ( reco::deltaR(*cluTP_p4, *trueEle_p4) < dR &&  (fabs(cluTP_p4->pt() - trueEle_p4->pt()) < 0.5*trueEle_p4->pt() ) ) {
+		dR=  reco::deltaR(*cluTP_p4, *trueEle_p4);
+		matchEle_p4 =  trueEle_p4;
+                gpIterMatch=gpIter;
+	      }
+	    }
+	  }
+	}
+	
+
+	if (  matchEle_p4 !=0 && dR<0.1) {
+
+	  	
+	  //float matchDPhi =  reco::deltaPhi (cluTP_p4->phi(), matchEle_p4->phi());
+	  float matchDPhi = normalizedPhi ( cluTP_p4->phi()- matchEle_p4->phi());
+	  //	if ( std::abs(matchDPhi)>CLHEP::pi) { matchDPhi = matchDPhi<0? (CLHEP::twopi) + matchDPhi : matchDPhi - CLHEP::twopi;}
+	  // float gpEta = etaTransformation( matchEle_p4->eta(), gpIterMatch->vz() );
+     	
+	  h_cluTPEtoverGenEt_->Fill ( cluTP_p4->pt()/ matchEle_p4->pt()); 
+	  h_dEta_cluTP_gen_ -> Fill(  cluTP_p4->eta()  - matchEle_p4->eta());
+	  h_dPhi_cluTP_gen_ -> Fill(matchDPhi);
+	  h_deltaR_cluTPGen_ -> Fill (dR);
+
+	}
+	
+
+      }  // if ( isGenParticleValid_ )
+
+      /*
```

Comment:
Delete commented out code, or justify with a comment why you still need to keep it